app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class TasksController < AuthenticationController
+  before_action :set_case_record
+  before_action :set_caseable
+  before_action :find_task
+  before_action :build_form, only: %i[edit update]
+
+  def show
+    respond_to do |format|
+      format.html { render template_for(:show) }
+    end
+  end
+
+  def edit
+    respond_to do |format|
+      format.html { render template_for(:edit) }
+    end
+  end
+
+  def update
+    respond_to do |format|
+      format.html do
+        if @form.update(task_params)
+          redirect_to @form.redirect_url, notice: "Task updated successfully"
```

Comment:
I assume at some point we'll want to customise this flash message but not a blocker to merging